mage/query-modules/python/temporal-graph-networks.md

```diff
@@ -0,0 +1,415 @@
+            np.concatenate([timestamps, timestamps]),
+        )
+
+        embeddings = self.tgn_net(graph_data)
+
+        ... process negative edges in a similar way
+        
+        self._process_current_batch(
+            sources, destinations, node_features, edge_features, edge_idxs, timestamps
+        )
+
+```
+Our `torch.nn.Module` is organized as follows:
+  * processing previous batches - if you follow *[research paper](https://arxiv.org/abs/2006.10637)* this will include a new calculation of messages collected for each node in the form of 
+    **message function**, aggregation of messages for each node in form of **message aggregator** and finally updating of each of the node's memory
+    with **memory updater**
+  * afterward we create a computation graph used by **graph attention layer** or **graph sum layer**
+  * final step includes processing of current batch, creating new **interaction or node events**, updating **raw message store** with new **events**
+
+Process repeats as we get new edges in batch, and after a batch is filled, we forward new edges again to **TGN** and so on...
+
+
+
+:::info
+
+This **MAGE** module is still in its early phase. We intended its use for only **learning** activities. The problem 
+with the current module is that we expect **you** to know when your stream will end so you can switch **TGN** mode
+from **train** to **eval**. After you do a switch, all incoming edges will be used for only **evaluation** and not
+any new **training**. If you wish to make it production-ready, because you like what you see :smile: be sure to either
+open **pull request** on our **[GitHub page](https://github.com/memgraph/mage)**, or drop us a comment on **[Discord](https://discord.gg/memgraph)**.
+Also, if you like what you saw, consider giving us a :star: so we can continue to do even better work. 
+
+:::
+   
+
+   
+
+| Trait               | Value                                                          |
+| ------------------- |----------------------------------------------------------------|
+| **Module type**     | <Highlight color="#FB6E00">**module**</Highlight>              |
+| **Implementation**  | <Highlight color="#FB6E00">**Python**</Highlight>              |
+| **Graph direction** | <Highlight color="#FB6E00">**directed**</Highlight>            |
+| **Edge weights**    | <Highlight color="#FB6E00">**weighted/unweighted**</Highlight> |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight>          |
+
+## Procedures
+
+### `set_params(learning_type, batch_size, num_of_layers, layer_type, memory_dimension, time_dimension, num_edge_features, num_node_features, message_dimension, num_neighbors, edge_message_function_type, message_aggregator_type, memory_updater_type, num_attention_heads, learning_rate, weight_decay)`
+
+#### Input:
+
+* `learning_type: str` ➡ "self_supervised" or "supervised" depending on if you want to predict edges or node labels
```

Comment:
Try bold or italic, or place variables in the brackets, so they are easily visible